### ISSUE TITLE
LFG conf option "Channel.RestrictedLfg"  not working

### ISSUE THREAD

**Issue by BarbzYHOOL** [azerothcore/azerothcore-wotlk#2756]
##### SMALL DESCRIPTION:

This option doesn't work `Channel.RestrictedLfg = 1`

##### EXPECTED BLIZZLIKE BEHAVIOUR:

I kinda expect this to work like this (but maybe i'm wrong):
- if set to 1, nobody is in LFG channel unless they queue for dungeon finder
- if set to 0, people are in LFG channel when they enter capitals (or everywhere if they have the other LFG config option enabled)

##### CURRENT BEHAVIOUR:

If I am not in looking for group channel (not in capital cities) and I use the dungeonfinder, i do not join the lookingforgroup channel
```
#
#    Channel.RestrictedLfg
#        Description: Restrict LookupForGroup channel to characters registered in the LFG tool.
#        Default:     1 - (Enabled, Allow join to channel only if registered in LFG)
#                     0 - (Disabled, Allow join to channel in any time)

Channel.RestrictedLfg = 1
```

Default option not working. 

Basically this option seems to do NOTHING at all.

##### STEPS TO REPRODUCE THE PROBLEM:

1. Edit the conf ^^
1. Queue for dungeon finder

##### BRANCH(ES):

master

##### AC HASH/COMMIT:

2eeab4c

**Comment by Viste** [comment]
According to code 0 also dont work ))

**Comment by xDevICCI** [comment]
Why was this issue reopened? It appears to be already fixed by PR #21145.
